polkaj-schnorrkel/src/main/java/io/emeraldpay/polkaj/merlin/TranscriptData.java

```diff
@@ -0,0 +1,65 @@
+package io.emeraldpay.polkaj.merlin;
+
+import java.math.BigInteger;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+
+/**
+ * A container class to simply hold all the necessary input data (label + messages) to construct the actual
+ * transcript on Rust's side. Think of this as "the bag of currently necessary arguments".
+ * The main idea of this class is to be easily portable to Rust using JNI.
+ * It makes no sense on its own on the Java side alone.
+ */
+public class TranscriptData {
+    // INTENTIONALITY: Those fields being of type ArrayList is essential for the JNI mappings
+
+    private final ArrayList<byte[]> domainSeparationLabel;
+
+    private final ArrayList<byte[]> labels;
+    private final ArrayList<byte[]> messages;
+
+    public TranscriptData(byte[] domainSeparationLabel) {
+        this.domainSeparationLabel = new ArrayList<>();
+        this.domainSeparationLabel.add(domainSeparationLabel);
+        this.labels = new ArrayList<>();
+        this.messages = new ArrayList<>();
+    }
+
+    public void appendMessage(String label, String message) {
+        appendMessage(label.getBytes(StandardCharsets.US_ASCII), message.getBytes(StandardCharsets.US_ASCII));
+    }
+
+    public void appendMessage(byte[] label, byte[] message) {
+        labels.add(label);
+        messages.add(message);
+    }
+
+    /**
+     * Appends a message with an u64 body.
+     * <br>
+     * NOTE: only the last 8 bytes of the BigInteger's byte[]
+     *       representation will be taken and reversed into little-endian order
+     * @param label the message label
+     * @param x the u64 to append
+     */
+    // TODO:
+    //  Maybe just remove this method and leave it up to the user...
+    //  We can't guarantee good enough type safety and obeying the representational contract is left to the developer,
+    //  which could lead to issues.
```

Comment:
Self explanatory. Still, leaving this convenience method in the public interface of this here `TranscriptData` (inspired by the original rust `schnorrkel`) might not be such a good idea in the Java world, since we don't have an unified `u64` type, so now it's up to the caller to understand my `NOTE:` in the javadoc... which stinks.

Reply:
I've removed it. We'll manage this u64 representation as a `byte[]` from Fruzhin.